Source/Scene/ModelUtility.js

```diff
@@ -33,6 +41,245 @@ define([
      */
     var ModelUtility = {};
 
+    function removeExtrasIfEmpty(object) {
+        if (!defined(object.extras)) {
+            return;
+        }
+
+        var pipeline = object.extras._pipeline;
+        if (!defined(pipeline)) {
+            return;
+        }
+
+        if (Object.keys(pipeline).length === 0) {
+            delete object.extras;
+        }
+    }
+
+    /**
+     * Removes empty extra._pipeline object from each object that can have extras in the glTF asset.
+     *
+     * @param {Object} gltf A javascript object containing a glTF asset.
+     * @returns {Object} The glTF asset with the added pipeline extras.
+     */
+    ModelUtility.removePipelineExtras = function(gltf) {
```

Comment:
It would also be good to get around the `.extras._pipeline` usage in `splitIncompatibleMaterials`, `processPbrMetallicRoughness` and `processModelMaterialsCommon`.